third_party/2and3/flask/app.pyi

```diff
@@ -0,0 +1,145 @@
+    def endpoint(self, endpoint: str) -> Callable[[Callable[..., _T]], Callable[..., _T]]: ...
+    def errorhandler(self, code_or_exception: Union[int, Type[Exception]]) -> Callable[[Callable[..., _T]], Callable[..., _T]]: ...
+    def register_error_handler(self, code_or_exception: Union[int, Type[Exception]], f: Callable[..., Any]) -> None: ...
+    def template_filter(self, name: Optional[Any] = ...): ...
+    def add_template_filter(self, f: Any, name: Optional[Any] = ...) -> None: ...
+    def template_test(self, name: Optional[Any] = ...): ...
+    def add_template_test(self, f: Any, name: Optional[Any] = ...) -> None: ...
+    def template_global(self, name: Optional[Any] = ...): ...
+    def add_template_global(self, f: Any, name: Optional[Any] = ...) -> None: ...
+    def before_request(self, f: Callable[[], _T]) -> Callable[[], _T]: ...
+    def before_first_request(self, f: Callable[[], _T]) -> Callable[[], _T]: ...
+    def after_request(self, f: Callable[[Response], Response]) -> Callable[[Response], Response]: ...
+    def teardown_request(self, f: Callable[[Optional[Exception]], _T]) -> Callable[[Optional[Exception]], _T]: ...
+    def teardown_appcontext(self, f: Callable[[Optional[Exception]], _T]) -> Callable[[Optional[Exception]], _T]: ...
+    def context_processor(self, f: Any): ...
+    def shell_context_processor(self, f: Any): ...
+    def url_value_preprocessor(self, f: Any): ...
+    def url_defaults(self, f: Any): ...
+    def handle_http_exception(self, e: Any): ...
+    def trap_http_exception(self, e: Any): ...
+    def handle_user_exception(self, e: Any): ...
+    def handle_exception(self, e: Any): ...
+    def log_exception(self, exc_info: Any) -> None: ...
+    def raise_routing_exception(self, request: Any) -> None: ...
+    def dispatch_request(self): ...
+    def full_dispatch_request(self): ...
+    def finalize_request(self, rv: Any, from_error_handler: bool = ...): ...
+    def try_trigger_before_first_request_functions(self): ...
+    def make_default_options_response(self): ...
+    def should_ignore_error(self, error: Any): ...
+    def make_response(self, rv: Any): ...
+    def create_url_adapter(self, request: Any): ...
+    def inject_url_defaults(self, endpoint: Any, values: Any) -> None: ...
+    def handle_url_build_error(self, error: Any, endpoint: Any, values: Any): ...
+    def preprocess_request(self): ...
+    def process_response(self, response: Any): ...
+    def do_teardown_request(self, exc: Any = ...) -> None: ...
+    def do_teardown_appcontext(self, exc: Any = ...) -> None: ...
+    def app_context(self): ...
+    def request_context(self, environ: Any): ...
+    def test_request_context(self, *args: Any, **kwargs: Any) -> ContextManager[RequestContext]: ...
+    def wsgi_app(self, environ: Any, start_response: Any): ...
+    def __call__(self, environ: Any, start_response: Any): ...
+
+    def __setattr__(self, name: str, value: Any): ...
+    def __getattr__(self, name: str): ...
```

Comment:
I would add a comment that these are not preset at runtime, but we add them since monkeypatching this class is quite common.